pkgs/build-support/alternatives/lapack/default.nix

```diff
@@ -54,10 +54,6 @@ stdenv.mkDerivation {
 '' + (if stdenv.hostPlatform.parsed.kernel.execFormat.name == "elf" then ''
   patchelf --set-soname liblapack${canonicalExtension} $out/lib/liblapack${canonicalExtension}
   patchelf --set-rpath "$(patchelf --print-rpath $out/lib/liblapack${canonicalExtension}):${lapackProvider}/lib" $out/lib/liblapack${canonicalExtension}
-'' else if stdenv.hostPlatform.isDarwin then ''
-  install_name_tool -id liblapack${canonicalExtension} \
```

Comment:
this look correct to me, does removing this fix the problem?

Reply:
No library identifiers should the absolute path to the dylib, with this the linker wil try to find `$PWD/liblapack.dylib` which obviously won't exsist in the numpy build, etc.

Reply:
Also rpaths are not really used on darwin so that won't do much.

Reply:
the important thing is that in mkl it gets the ${mkl}/lib directory in the RPATH so that it can use dlopen to find the proper. Also ideally the name would be `liblapack.dylib` so that you could provide a custom value DYLD_LIBRARY_PATH, but that may not work as I expected.

Reply:
I have no idea how rpath on a dylib behaves, but I'm not sure that dlopen needs or uses the library identifier.